### PR TITLE
docs: add dynamic badge to configuration options

### PR DESCRIPTION
**Pull request by bmorelli25**
## What does this pull request do?

This PR adds the `Dynamic` badge to supported configurations. I defined "supported configurations" as options that are supported by the [Apm app](https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/common/agent_configuration/setting_definitions/general_settings.ts). 

- [x] `api_request_size`
- [x] `api_request_time`
- [x] `capture_body`
- [x] `capture_headers`
- [x] `log_level`
- [x] `span_frames_min_duration`
- [x] `transaction_max_spans`
- [x] `transaction_sample_rate`

There is a discrepancy between the list above, and the list created by @estolfo in https://github.com/elastic/apm-agent-ruby/pull/741: `recording` doesn't seem to be a valid option yet.

## Documentation preview

http://apm-agent-ruby_760.docs-preview.app.elstc.co/diff

## Screenshot

Screen Shot 2020-04-08 at 4 56 48 PM

## Related issues

For https://github.com/elastic/apm/issues/232
For https://github.com/elastic/observability-dev/issues/639

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -8,27 +8,46 @@ endif::[]
 == Configuration
 
 There are several ways to configure how Elastic APM behaves.
-
-The recommended way to configure Elastic APM is to create a file
-`config/elastic_apm.yml` and specify options in there:
+The recommended way is to specify options in a `config/elastic_apm.yml` file:
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:11]: Is this still the recommend way?

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -192,6 +205,9 @@ went wrong.
 [float]
 [[config-api-request-size]]
 ==== `api_request_size`
+
+<<dynamic-configuration, image:./images/dynamic-config.svg[] >>
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:209]: These badges are also links to the <<dynamic-configuration>> section

**Comment by bmorelli25** [comment]
I'm going to merge this, but I'm happy to make changes later if you have feedback.

**Comment by estolfo** [comment]
Sorry @bmorelli25 we had a holiday in Germany last Friday until today so I didn't get a chance to review this. I'll review it and open a PR if I see anything that should be changed.

**Pull request merged**